### PR TITLE
Update parser services + fix wrapped rule bug

### PR DESCRIPTION
**Pull request by hjdivad**
- Add pathOf to parser services
- Fix typeInfo type stack bug

### CODE REVIEW

File: projects/graphql/example/tests/custom-rule-test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      {
+        code: `
+        query {
+          bookstores {
+            books {
+              title
+            }
+          }
+        }
+      `,
```
Comment (scalvert) [inline projects/graphql/example/tests/custom-rule-test.js:65]: Are there any other permutations you think we should be testing for here?

**Review by scalvert — APPROVED** [review]

File: projects/graphql/example/tests/custom-rule-test.js (hunk 1 above)
Comment (hjdivad) [reply projects/graphql/example/tests/custom-rule-test.js:65]: hmm there's a couple things we might as well test

1. paths that use variables (shouldn't affect anything but might as well have a test for it)
2. paths that cross fragment boundaries (inline + in-doc)

**Review by stefanpenner — APPROVED** [review]

File: projects/graphql/eslint-plugin-graphql/lib/graphql-rule-adapter.js [hunk 2]
```diff
@@ -62,12 +62,16 @@ module.exports = function adapt(rule) {
       );
       const rules = rule(validationContext);
 
-      const callbacks = { enter() {}, exit() {} };
+      const callbacks = { enter() {}, leave() {} };
 
       wrappedRules['*'] = function (node) {
         typeInfo.enter(context.parserServices.correspondingNode(node));
         callbacks.enter(node);
-        callbacks.exit(node);
```
[the remark this answers is not in the document]
Comment (hjdivad) [reply projects/graphql/eslint-plugin-graphql/lib/graphql-rule-adapter.js:70]: @stefanpenner i fixed this but i don't know if the `callbacks` stuff is actually covered by our tests -- the type stack *is* covered by a test with sibling fields + selectionsets

[the remark this answers is not in the document]
Comment (stefanpenner) [reply projects/graphql/eslint-plugin-graphql/lib/graphql-rule-adapter.js:70]: I totally got sloppy for this scenario, this gives me further evidence that test all the things is actually important.

File: projects/graphql/example/tests/custom-rule-test.js (hunk 1 above)
Comment (hjdivad) [reply projects/graphql/example/tests/custom-rule-test.js]: @scalvert I've updated the implementation and added tests to cover more cases; thoughts?

Comment (scalvert) [reply projects/graphql/example/tests/custom-rule-test.js]: 👍 tests look good. Thanks for adding.

**Pull request merged**